### ISSUE TITLE
Bug: ZSH completion is not working for Java

### ISSUE THREAD

**Issue by sureshg** [sdkman/sdkman-cli#985]
**Bug report**
Pressing tab after entering `sdk install java` throws the following error on MacOS
```
 $ sdk install java  <TAB> __sdkman_determine_current_version:8: bad math expression: operator expected at `ea.2.lm-op...'                                    

_sdk:35: substring expression: -4 < 5
```

**To reproduce**
Press tab after entering `sdk install java`

**System info**
```
MacOS: 11.5.2
zsh 5.8 (x86_64-apple-darwin20.0)
SDKMAN 5.12.4
```

**Comment by marc0der** [comment]
Looking at the zsh completion script, the offending line seems to be this:

```
compadd -X $'Installable Versions of java:\n' -- "${${${${${(f)$(__sdkman_list_versions "${words[3]}")}[@]:5:-4}[@]:#* | (local only|installed ) | *}[@]##* |            | }[@]%%+( )}"
```

I'm not seeing the same issue on Linux, but I think the new shape of the list view might have broken this crazy regex. Anyone who is able to help, please let us know. Alternatively, I could roll back the list view.

**Comment by cy6erGn0m** [comment]
Have the same error under Linux but with slightly different error (because of zsh, I think)

>  __sdkman_determine_current_version:8: bad floating point constant
> _sdk:35: substring expression: -3 < 5

**Comment by cy6erGn0m** [comment]
Don't get why but for me it does evaluate version string as a floating point number...

According to xtrace:

```
+__sdkman_path_contains:6> echo true
+__sdkman_determine_current_version:4> present=true 
+__sdkman_determine_current_version:5> [[ true == true ]]
+__sdkman_determine_current_version:6> [[ $PATH -regex-match ${SDKMAN_CANDIDATES_DIR}/${candidate}/([^/]+)/bin ]]
+__sdkman_determine_current_version:7> [[ true == true ]]
+__sdkman_determine_current_version:8> CURRENT=11.0.12-open __sdkman_determine_current_version:8: bad floating point constant
```

**Comment by cy6erGn0m** [comment]
After upgrading zsh from 5.6 to 5.8 it stopped crashing and does work but it looks weird:

```
Installable Versions of java:
 
\================================================================================  17.ea.3.pma-open                                                                 
\ \ \ \ \$\ sdk\ install\ java                                                     17-sapmchn                                                                       
\ \ \ \ \$\ sdk\ install\ java\ \[TAB\]                                            18.ea.15-open                                                                    
11.0.10-open                                                                       18.ea.16-open                                                                    
11.0.11.hs-adpt                                                                    18.ea.2.lm-open                                                                  
11.0.11.j9-adpt                                                                    19.3.6.r11-grl                                                                   
11.0.11-open                                                                       19.3.6.r8-grl                                                                    
11.0.12.7.1-amzn                                                                   20.3.2.r8-grl                                                                    
11.0.12.7.1-ms                                                                     20.3.3.0-mandrel                                                                 
11.0.12.fx-librca                                                                  20.3.3.r11-grl                                                                   
11.0.12.fx-zulu                                                                    20.3.3.r8-grl                                                                    
11.0.12-librca                                                                     21.0.0.2-nik                                                                     
11.0.12-sapmchn                                                                    21.1.0.r8-grl                                                                    
11.0.12-sem                                                                        21.1-nik                                                                         
11.0.12-tem                                                                        21.2.0.0-mandrel                                                                 
11.0.12-zulu                                                                       21.2.0.r11-grl                                                                   
11.0.2-open                                                                        21.2.0.r16-grl                                                                   
11.0.9.4-albba                                                                     21.2.0.r8-grl                                                                    
11.0.9-trava                                                                       21.2-nik                                                                         
16.0.1.hs-adpt                                                                     6.0.119-zulu                                                                     
16.0.1.j9-adpt                                                                     7.0.312-zulu                                                                     
16.0.2.7.1-amzn                                                                    8.0.232-trava                                                                    
16.0.2.7.1-ms                                                                      8.0.265-open                                                                     
16.0.2.fx-librca                                                                   8.0.282-open                                                                     
16.0.2.fx-zulu                                                                     8.0.292.j9-adpt                                                                  
16.0.2-librca                                                                      8.0.292-open                                                                     
16.0.2-open                                                                        8.0.302.fx-librca                                                                
16.0.2-sapmchn                                                                     8.0.302.fx-zulu                                                                  
16.0.2-sem                                                                         8.0.302-librca                                                                   
16.0.2-tem                                                                         8.0.302-open                                                                     
16.0.2-zulu                                                                        8.0.302-sem                                                                      
17.0.0.35.1-amzn                                                                   8.0.302-tem                                                                      
17.0.0.fx-librca                                                                   8.0.302-zulu                                                                     
17.0.0.fx-zulu                                                                     8.302.08.1-amzn                                                                  
17.0.0-librca                                                                      8.5.5-albba                                                                      
17.0.0-oracle                                                                      Omit\ Identifier\ to\ install\ default\ version\ 17.0.0-tem:                     
17.0.0-tem                                                                         Use\ TAB\ completion\ to\ discover\ available\ versions                          
17.0.0-zulu                                                                                                                                                 
```

**Comment by marc0der** [comment]
@cy6erGn0m that's exactly what I was seeing on Linux. This is due to the invalid regex that needs updating for which I requested some help from the community.

**Comment by helpermethod** [comment]
This has been fixed in #1002. Since then, no regression could be observed.
